node/gum/proc-macro/src/types.rs

```diff
@@ -0,0 +1,268 @@
+pub(crate) enum FormatMarker {
+	Questionmark(Token![?]),
+	Percentage(Token![%]),
+	None,
+}
+
+impl Parse for FormatMarker {
+	fn parse(input: ParseStream) -> Result<Self> {
+		let lookahead = input.lookahead1();
+		if lookahead.peek(Token![?]) {
+			input.parse().map(Self::Questionmark)
+		} else if lookahead.peek(Token![%]) {
+			input.parse().map(Self::Percentage)
+		} else {
+			Ok(Self::None)
+		}
+	}
+}
+
+impl ToTokens for FormatMarker {
+	fn to_tokens(&self, tokens: &mut TokenStream) {
+		tokens.extend(match self {
+			Self::Percentage(p) => p.to_token_stream(),
+			Self::Questionmark(q) => q.to_token_stream(),
+			Self::None => TokenStream::new(),
+		})
+	}
+}
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub(crate) struct ValueWithAliasIdent {
+	pub alias: Ident,
+	pub eq: Token![=],
+	pub marker: FormatMarker,
+	pub expr: syn::Expr,
+}
+
+impl Parse for ValueWithAliasIdent {
+	fn parse(input: ParseStream) -> Result<Self> {
+		Ok(Self {
+			alias: input.parse()?,
+			eq: input.parse()?,
+			marker: input.parse()?,
+			expr: input.parse()?,
+		})
+	}
+}
+
+impl ToTokens for ValueWithAliasIdent {
+	fn to_tokens(&self, tokens: &mut TokenStream) {
+		let alias = &self.alias;
+		let eq = &self.eq;
+		let marker = &self.marker;
+		let expr = &self.expr;
+		tokens.extend(quote! {
+			#alias #eq #marker #expr
+		})
+	}
+}
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+
+pub(crate) struct ValueWithFormatMarker {
+	pub marker: FormatMarker,
+	pub ident: Ident,
+}
+
+impl Parse for ValueWithFormatMarker {
+	fn parse(input: ParseStream) -> Result<Self> {
+		Ok(Self { marker: input.parse()?, ident: input.parse()? })
+	}
+}
+
+impl ToTokens for ValueWithFormatMarker {
+	fn to_tokens(&self, tokens: &mut TokenStream) {
+		let marker = &self.marker;
+		let expr = &self.ident;
+		tokens.extend(quote! {
+			#marker #expr
+		})
+	}
+}
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+
+pub(crate) enum Value {
+	Alias(ValueWithAliasIdent),
+	Value(ValueWithFormatMarker),
+}
+
+impl Value {
+	pub fn as_ident(&self) -> &Ident {
+		match self {
+			Self::Alias(alias) => &alias.alias,
+			Self::Value(value) => &value.ident,
+		}
+	}
+}
+
+impl Parse for Value {
+	fn parse(input: ParseStream) -> Result<Self> {
+		if input.fork().parse::<ValueWithAliasIdent>().is_ok() {
+			input.parse().map(Self::Alias)
+		} else if input.fork().parse::<ValueWithFormatMarker>().is_ok() {
+			input.parse().map(Self::Value)
+		} else {
+			Err(syn::Error::new(Span::call_site(), "Neither value nor aliased value."))
+		}
+	}
+}
+
+impl ToTokens for Value {
+	fn to_tokens(&self, tokens: &mut TokenStream) {
+		tokens.extend(match self {
+			Self::Alias(alias) => quote! { #alias },
+			Self::Value(value) => quote! { #value },
+		})
+	}
+}
+
+#[derive(Debug, Clone)]
+pub(crate) struct Args {
+	pub target: Option<Target>,
+	pub comma: Option<Token![,]>,
+	pub values: Punctuated<Value, Token![,]>,
+	// TODO use `parse_fmt_str:2.0.0`
+	// instead to sanitize
```

Comment:
It depends, right now the evaluation of the fmt literal Str is done by `tracing::event!`, I have yet to check if the error message is displayed correctly, if so this can be deleted, otherwise it will be addressed in this PR

Reply:
Deferred.